pkg/storageincentives/agent.go

```diff
@@ -254,35 +247,37 @@ func (a *Agent) start(blockTime time.Duration, blocksPerRound, blocksPerPhase ui
 		}
 
 		// write the current phase only once
-
 		if currentPhase != prevPhase {
 
 			a.metrics.CurrentPhase.Set(float64(currentPhase))
 
 			a.logger.Info("entering phase", "phase", currentPhase.String(), "round", round, "block", block)
 
+			a.nodeState.SetCurrentEvent(currentPhase, round, block)
+
 			phaseEvents.Publish(currentPhase)
 			if currentPhase == claim {
 				phaseEvents.Publish(sample) // trigger sample along side the claim phase
 			}
 		}
 
 		prevPhase = currentPhase
-
 		mtx.Unlock()
+
 	}
 }
 
 func (a *Agent) reveal(ctx context.Context, storageRadius uint8, sample, obfuscationKey []byte) error {
 	a.metrics.RevealPhase.Inc()
-	err := a.contract.Reveal(ctx, storageRadius, sample, obfuscationKey)
+	fee, err := a.contract.Reveal(ctx, storageRadius, sample, obfuscationKey)
 	if err != nil {
 		a.metrics.ErrReveal.Inc()
 	}
+	a.nodeState.AddFee(fee)
```

Comment:
if err is not nil, we should return the error, if err is nil, only then we add the fee